BiometricLocker/BiometricLocker.swift

```diff
@@ -80,16 +99,35 @@ final public class BiometricLocker {
         return false
     }
 
-    public enum When {
-        case now
-        case afterTimeAllowance
-        case afterTimeInterval(TimeInterval)
+    /// Store the NotificationCenter observers for deallocation.
+    private var notificationObservers = [NSObjectProtocol]()
+
+    public init() {
+        self.notificationObservers.append(NotificationCenter.default.addObserver(forName: .UIApplicationDidEnterBackground, object: nil, queue: .main) { _ in
+            // If app goes into background, we start the clocks to lock the app.
+            self.lock()
+        })
+
+        self.notificationObservers.append(NotificationCenter.default.addObserver(forName: .UIApplicationWillTerminate, object: nil, queue: .main) { _ in
+            // If the app is killed, lock it instantly.
+            self.lock(.now)
+        })
     }
 
-    /// Locks the app.
-    ///
-    /// - Parameter date: Tells the locker when the app was sent to the background (or any other situation, when applicable), so that we can calculate if the app should be locked.
-    public func lock(_ when: When = .afterTimeAllowance) {
+    deinit {
+        self.notificationObservers.forEach { observer in
+            NotificationCenter.default.removeObserver(observer)
+        }
```

Comment:
since this isn't a singleton, won't this cancel the observation as soon as the locker goes out of scope?